basket/base/rq.py

```diff
@@ -0,0 +1,231 @@
+import os
+import random
+import re
+import traceback
+from time import time
+
+from django.conf import settings
+
+import redis
+import requests
+import sentry_sdk
+from django_statsd.clients import statsd
+from rq import Callback, Retry, SimpleWorker
+from rq.queue import Queue
+from rq.serializers import JSONSerializer
+from silverpop.api import SilverpopResponseException
+
+from basket.base.exceptions import RetryTask
+from basket.news.backends.common import NewsletterException
+
+# don't propagate and don't retry if these are the error messages
+IGNORE_ERROR_MSGS = [
+    "INVALID_EMAIL_ADDRESS",
+    "InvalidEmailAddress",
+    "An invalid phone number was provided",
+    "No valid subscribers were provided",
+    "There are no valid subscribers",
+    "email address is suppressed",
+    "invalid email address",
+]
+# don't propagate and don't retry if these regex match the error messages
+IGNORE_ERROR_MSGS_RE = [re.compile(r"campaignId \d+ not found")]
+# Exceptions we allow to retry, all others will abort retries.
+EXCEPTIONS_ALLOW_RETRY = [
+    IOError,
+    NewsletterException,
+    requests.RequestException,
+    RetryTask,
+    SilverpopResponseException,
+]
+
+
+# Our cached Redis connection.
+_REDIS_CONN = None
+
+
+def get_redis_connection(url=None, force=False):
+    """
+    Get a Redis connection.
+
+    Expects a URL including the db, or defaults to `os.environ["RQ_URL"]`.
+
+    Call example:
+        get_redis_connection("redis://localhost:6379/0")
+
+    """
+    global _REDIS_CONN
+
+    if url is None:
+        url = os.environ.get("RQ_URL")
```

Comment:
any reason you're not using Django settings here?

Reply:
Not really, other than that's the pattern that already existed in the settings file. We're creating the `RQ_URL` env variable from the `REDIS_URL`. But we could also make it a django setting rather than an env variable. I'll add it to the task in the PR to clean this up.